wordpress-mobile/WordPress-Android
# Task/adjust choose topics

Users were complaining that they can't access the following tab unless they select an interests. We decided to rework when and where is the select interest screen shown.

Props to @zwarm for kicking of these changes

Previously, if the user wasn't following any tags, the select interest screen was shown over the Reader fragment. Users couldn't access the Reader at all unless they started following a topic.
After this change, users can access Reader no matter if they follow any topics. An empty layout is shown on the discover tab if the user isn't following any topics. The select interest screen is shown only after the user clicks on a button on the empty layout.

Responsibilities:
ReaderFragment/VM - fragment which contains a container for the select interest screen. It observes open/close interest screen events and acts accordingly.
ReaderDiscoverFragment/VM - Checks whether the user follows any tags and displays either discover feed or empty layout with "select interests" button which leads to the "select interests" screen.
ReaderInterestFragment/VM - The select interest screen itself.


<img src="https://user-images.githubusercontent.com/2261188/96120687-34c17b80-0eef-11eb-8235-0540325f6391.gif" width="250px" />

<img src="https://user-images.githubusercontent.com/2261188/96120675-2ffcc780-0eef-11eb-9507-11f6bafba96d.gif" width="250px" />



To test:
1. Unfollow all tags
2. Restart the app
3. Open Reader
4. Open Discover tab
5. Notice an empty layout is shown
6. Click on the button
7. Select a few interests
8. Click on done and verify that the discover feed is refreshed automatically
9. Go to Following Tab
10. Click on the gear icon
11. Unfollow all tags
12. Go back to discover tab
13. Notice the empty layout is shown again

+ anything else you can think of

PR submission checklist:
